samples/volumes/example5.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import kfp.dsl as dsl
+
+
+@dsl.pipeline(
+    name="Example 5",
+    description="The fifth example of the design doc."
+)
+def example5(rok_url):
+    vol1 = dsl.PipelineVolume(
```

Comment:
This class mingles volume spec and creating volume operation together, which is not ideal to me. Can we make them as separate classes, for example:

PipelineVolume is served as a data object which contains the volume spec.
PersistentVolumeClaimOp is similar with ContainerOp which serves as creating a pvc in the pipeline.
PipelineVolume can be an output of PersistentVolumeClaimOp.

Later, we might refactor the code that PipelineVolume derives from k8s volume spec, and PersistentVolumeClaimOp derives from K8sResourceOp which can be used to create general k8s resources.

WDYT?

Reply:
@hongye-sun 
> This class mingles volume spec and creating volume operation together, which is not ideal to me. Can we make them as separate classes, for example:
>
> PipelineVolume is served as a data object which contains the volume spec.
> PersistentVolumeClaimOp is similar with ContainerOp which serves as creating a pvc in the pipeline.
> PipelineVolume can be an output of PersistentVolumeClaimOp.

Yes! This was a great comment.
We went through this with @elikatsis , and tried to iterate on this approach.
We have verified with actual code that this indeed improves the design even further.

Essentially what you are saying is:
Let's separate the data object, which refers to the data being passed from step to step, from the *operation*, the DAG step that creates the object.

This wil work. First have a generic way to create a K8s resource, similar to `ContainerOp`, then inherit from it so we can have an `Op` that creates PVCs, then have this `Op` produce a data objct that is the `PipelineVolume`, the volume itself, ready to be consumed by next steps.

> Later, we might refactor the code that PipelineVolume derives from k8s volume spec, and PersistentVolumeClaimOp derives from K8sResourceOp which can be used to create general k8s resources.
>
> WDYT?

This was a great idea, I'll be amending the design doc, and we have verified this works, with real code.